addon/components/mdc-drawer.js

```diff
@@ -0,0 +1,123 @@
+      throw new Ember.Error('Cannot be persistent and permanent or temporary');
+    }
+    return persistent;
+  }),
+  isTemporary: computed('permanent', 'persistent', 'temporary', function () {
+    const {permanent, persistent, temporary} = getProperties(this, 'permanent', 'persistent', 'temporary');
+    if (temporary && (permanent || persistent)) {
+      throw new Ember.Error('Cannot be temporary and permanent or persistent');
+    }
+    return temporary;
+  }),
+  //endregion
+
+  //region Methods
+  updateOpenness() {
+    const foundation = get(this, 'foundation');
+    if (!foundation) { return; }
+    const open = get(this, 'open');
+    if (foundation.isOpen() && !open) {
+      foundation.close();
+    }
+    if (!foundation.isOpen() && open) {
+      foundation.open();
+    }
+  },
+  /**
+   * @returns {MDCPersistentDrawerFoundation|MDCTemporaryDrawerFoundation}
+   */
+  createFoundation() {
+    const {permanent, persistent} = getProperties(this, 'permanent', 'persistent');
+    if (permanent) {
+      return { init() {}, destroy() {} };
+    }
+    const Foundation = persistent ? MDCPersistentDrawerFoundation : MDCTemporaryDrawerFoundation;
+    const {FOCUSABLE_ELEMENTS, ITEMS_SELECTOR, DRAWER_SELECTOR} = Foundation.strings;
+    const adapter = {
+      addClass: className => get(this, 'mdcClasses').addObject(className),
+      removeClass: className => get(this, 'mdcClasses').removeObject(className),
+      hasClass: className => get(this, 'element.classList').contains(className),
+      registerInteractionHandler: (type, handler) => this.registerMdcInteractionHandler(type, handler),
+      deregisterInteractionHandler: (type, handler) => this.deregisterMdcInteractionHandler(type, handler),
+      registerDrawerInteractionHandler: (type, handler) => this.registerMdcInteractionHandler(type, handler),
+      deregisterDrawerInteractionHandler: (type, handler) => this.deregisterMdcInteractionHandler(type, handler),
+      registerTransitionEndHandler: (handler) => this.registerMdcInteractionHandler('transitionend', handler),
+      deregisterTransitionEndHandler: (handler) => this.deregisterMdcInteractionHandler('transitionend', handler),
+      registerDocumentKeydownHandler: (handler) => this.registerMdcInteractionHandler('keydown', handler),
+      deregisterDocumentKeydownHandler: (handler) => this.deregisterMdcInteractionHandler('keydown', handler),
+      getDrawerWidth: () => get(this, 'element').offsetWidth,
+      setTranslateX: (value) => this.setStyleFor('mdcStyles', 'translateX', `${value}px`),
+      saveElementTabState: (el) => set(this, 'previousTab', el.tabIndex),
+      restoreElementTabState: (el) => el.tabIndex = get(this, 'previousTab'), //WTF
+      makeElementUntabbable: (el) => el.tabIndex = -1,
+      notifyOpen: () => set(this, 'open', true),
+      notifyClose: () => set(this, 'open', false),
+      isRtl: () => window.getComputedStyle(get(this, 'element')).getPropertyValue('direction') === 'rtl',
```

Comment:
For safety, it may be a good idea to use `utils/get-element-property` here. Example:

https://github.com/secondstreet/ember-material-components-web/blob/16cd18316f7f29be05b35dce6bebf7a8aa9d89cf/addon/components/mdc-radio.js#L65